src/core/CoreSession.jl

```diff
@@ -5,61 +5,146 @@ const PULSE_INTERVAL_MILLIS = 5000
-        session_id = res.session_id
-        transactions = Array{Union{Nothing,<:AbstractCoreTransaction},1}(nothing,0)
+        session_id =  res_id
+        transactions = Dict{UUID,AbstractCoreTransaction}()
         is_open = true
 
-        result = CoreSession(client, database, session_id, transactions, type, options, is_open, networkLatencyMillis)
+        t = Controller(true, 0.5)
+
+        result = CoreSession(client, database, session_id, transactions, type, ReentrantLock() ,options, is_open, networkLatencyMillis, t, request_timout)
+
+        make_pulse_request(result, t)
+
+        # the following transaction is useless for the construction of the the Session
+        # and is only to initialize the first compilation of the ProtoBuf machinary
+        # underneath the transaction. In the environment here it is possible to control
+        # the tick of the pulse request a close as can be. If somebody find a way to
+        # avoid the long starting time of the transaction this can be removed.
+        trans = nothing
+        try
+            trans = transaction(result, Proto.Transaction_Type[:READ])
+        catch ex
+            @info "First attempt for transaction done and not successful"
+        finally
+        end
+        if trans !== nothing
+            close(trans)
+        end
 
-        @async start_pulse(result, (PULSE_INTERVAL_MILLIS / 1000))
+        t.duration_in_seconds =  (PULSE_INTERVAL_MILLIS / 1000) - 0.5
 
         return result
     # catch ex
     #     throw(GraknClientException("Error construct a CoreSession",ex))
     # end
 end
 
-function start_pulse(session::T, pulse_time::Number) where {T<:AbstractCoreSession}
-    while session.isOpen
-        make_pulse_request(session)
-        sleep(pulse_time - 1)
+"""
+function make_pulse_request(session::T) where {T<:AbstractCoreSession}
+    This function make a pulse request to keep the session alive.
+"""
+function make_pulse_request(session::T, controller::Controller) where {T<:AbstractCoreSession}
+
+    @async begin
+            while controller.running
+                    try
+                        pulsreq = SessionRequestBuilder.pulse_req(session.sessionID)
+                        req_result, status = session_pulse(session.client.core_stub.blockingStub, gRPCController() , pulsreq)
+                        result = grpc_result_or_error(req_result,status, result->result)
+                        @info "Time: $(Dates.now())"
+
+                        if result.alive === false
+                            close(session)
+                            @info "$session is closed"
+                        end
+                    catch ex
+                        @info "make_pulse_request show's an error \n
+                        $ex "
+                        close(session)
+                    finally
+                    end
+                    sleep(controller.duration_in_seconds)
+            end
+        end
+end
```

Comment:
- Too much indentation. 
- At line 95, you can use [triple-quoted literals](https://docs.julialang.org/en/v1/manual/strings/#Triple-Quoted-String-Literals)

Can we also use `@debug` rather than `@info` for these pulse requests? It's somewhat annoying when you try to use the REPL and it keeps disturbing the console.